src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -2436,18 +2436,22 @@ def import_ssl_cert(cmd, resource_group_name, name, key_vault, key_vault_certifi
     kv_resource_group_name = kv_id_parts['resource_group']
     kv_subscription = kv_id_parts['subscription']
 
-    if kv_subscription.lower() != client.config.subscription_id.lower():
-        diff_subscription_client = get_mgmt_service_client(cmd.cli_ctx, ResourceType.MGMT_APPSERVICE,
-                                                           subscription_id=kv_subscription)
-        ascs = diff_subscription_client.app_service_certificate_orders.list()
+    cloud_type = cmd.cli_ctx.cloud.name
+    if cloud_type.lower() == "AzureCloud".lower():
```

Comment:
Is Azure Cloud public? if so this logic is incorrect. Note: the change expected is
client.app_service_certificate_orders -> is supported on public only so making this call should be avoided on all non-public clouds.